packages/bolt-integration/src/bolt-payment-strategy.ts

```diff
@@ -0,0 +1,370 @@
+            .getPaymentMethodOrThrow(options.methodId);
+
+        const { initializationData, config } = paymentMethod;
+        const { publishableKey, developerConfig, embeddedOneClickEnabled } =
+            initializationData || {};
+        const { testMode } = config;
+
+        if (!publishableKey) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this.boltClient = await this.boltScriptLoader.loadBoltClient(
+            publishableKey,
+            testMode,
+            developerConfig,
+        );
+
+        this.useBoltClient = !embeddedOneClickEnabled;
+        this.useBoltEmbedded = !!embeddedOneClickEnabled;
+
+        if (this.useBoltEmbedded) {
+            if (!containerId) {
+                throw new InvalidArgumentError(
+                    'Unable to initialize payment because "options.bolt.containerId" argument is not provided.',
+                );
+            }
+
+            if (!onPaymentSelect) {
+                throw new InvalidArgumentError(
+                    'Unable to initialize payment because "options.bolt.onPaymentSelect" argument is not provided.',
+                );
+            }
+
+            this.boltEmbedded = await this.boltScriptLoader.loadBoltEmbedded(
+                publishableKey,
+                testMode,
+                developerConfig,
+            );
+
+            this.mountBoltEmbeddedField(containerId);
+            onPaymentSelect(await this.hasBoltAccount());
+        }
+    }
+
+    deinitialize(): Promise<void> {
+        this.embeddedField?.unmount();
+
+        this.boltClient = undefined;
+        this.boltEmbedded = undefined;
+
+        return Promise.resolve();
+    }
+
+    finalize(): Promise<void> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<void> {
+        this.setExtraItemsForAnalytics();
+
+        const { payment, ...order } = payload;
+        const { methodId, paymentData } = payment || {};
+        let paymentPayload;
+
+        if (!payload.payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        if (!methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!paymentData || !isBoltPaymentData(paymentData)) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        await this.paymentIntegrationService.submitOrder(order, options);
+
+        if (this.useBoltClient) {
+            paymentPayload = await this.getBoltClientPaymentPayload(methodId, paymentData, options);
+        } else if (this.useBoltEmbedded) {
+            paymentPayload = await this.getBoltEmbeddedPaymentPayload(methodId, paymentData);
+        } else {
+            paymentPayload = await this.getBoltFullCheckoutPaymentPayload(methodId, paymentData);
+        }
```

Comment:
[Optional] We can create another method like `getPaymentData` to reduce amount of code in execute method. Hope it will be simpler to read.

What do you think?

Reply:
I think no need to do this, here we can see that we have three different data sources for different execution methods.
So propose to left as it is.